yarn-project/pxe/src/kernel_prover/private_inputs_builders/build_private_kernel_reset_hints.ts

```diff
@@ -47,22 +48,25 @@ function getNullifierReadRequestHints(
   return buildNullifierReadRequestHints({ getNullifierMembershipWitness }, nullifierReadRequests, nullifiers);
 }
 
-async function getMasterNullifierSecretKeys(
-  nullifierKeyValidationRequests: Tuple<
-    ScopedNullifierKeyValidationRequest,
-    typeof MAX_NULLIFIER_KEY_VALIDATION_REQUESTS_PER_TX
-  >,
+async function getMasterSecretKeysAndAppKeyGenerators(
```

Comment:
Is this really hints? Usually the hints have been stuff like merkle paths where it was easily validated that not lying etc, but what actually happens if I provide "wrong" paths here with a malicious oracle?

Reply:
The values are used to verify that the keys in `KeyValidationRequest` can be derived using the hints (sk_m and sk_app generator index). So if you provide wrong hints here the proof would fail.